aggregator_core/src/datastore.rs

```diff
@@ -1917,6 +1920,37 @@ impl<C: Clock> Transaction<'_, C> {
         )
     }
 
+    /// Deletes an aggregation job, including any contained report_aggregations.
+    #[tracing::instrument(skip(self), err)]
+    pub async fn delete_aggregation_job(
+        &self,
+        task_id: &TaskId,
+        aggregation_job_id: &AggregationJobId,
+    ) -> Result<(), Error> {
+        // Deleting report_aggregations is achieved through ON DELETE CASCADE.
+        let stmt = self.prepare_cached(
+            "DELETE FROM aggregation_jobs
+            USING tasks
+            WHERE tasks.id = aggregation_jobs.task_id
+              AND tasks.task_id = $1
+              AND aggregation_jobs.aggregation_job_id = $2
+              AND UPPER(aggregation_jobs.client_timestamp_interval) >= COALESCE($3::TIMESTAMP - tasks.report_expiry_age * '1 second'::INTERVAL, '-infinity'::TIMESTAMP)"
```

Comment:
This is likely not important to the overall correctness of the system, but we are pretending that anything past the logical expiry time already does not exist. So we'd want to get `MutationTargetNotFound` if we attempt to delete an expired aggregation job, rather than success.